### ISSUE TITLE
remove CPLEX dependency

### ISSUE THREAD

**Issue by karlnapf** [shogun-toolbox/shogun#2791]
and investigate how we can replace it with another open-source LP solver

Useful:
 * http://www.statistik.tuwien.ac.at/forschung/CS/CS-2012-1complete.pdf
 * http://en.wikipedia.org/wiki/List_of_optimization_software

**Comment by cdawei** [comment]
Hi, @karlnapf I want to work on this issue, could you please guide me?

**Comment by cdawei** [comment]
Hi, @karlnapf the Osi (Open Solver Interface) from COIN-OR seems a good candidate.
https://projects.coin-or.org/Osi/

**Comment by erip** [comment]
Google has the [operations research tools](https://github.com/google/or-tools) repo. Might be nice.

**Comment by theaverageguy** [comment]
@karlnapf Any call on what else we can use? I will work on this.

**Comment by karlnapf** [comment]
I dont think so. This needs discussion before started. So make sure to ask in IRC. @sonney2k @lisitsyn @vigsterkr will have good comments

**Comment by souravsingh** [comment]
We can make use of GLPK or dlib. I am assuming shogun already makes use of GLPK.
